skbio/sequence/_genetic_code.py

```diff
@@ -8,588 +8,776 @@
 
 from __future__ import absolute_import, division, print_function
 
-import re
-from collections import defaultdict
+import numpy as np
+from future.builtins import range
 
+from skbio.util._decorator import classproperty
 from skbio._base import SkbioObject
-from skbio.sequence import Protein, InvalidCodonError, GeneticCodeInitError
-
-# py3k compatibility
-try:
-    from string import maketrans
-except ImportError:
-    maketrans = str.maketrans
-
-_dna_trans = maketrans('TCAG', 'AGTC')
-
-
-def _simple_rc(seq):
-    """simple reverse-complement: works only on unambiguous uppercase DNA"""
-    return seq.translate(_dna_trans)[::-1]
+from skbio.sequence import Protein, RNA
+from skbio.sequence._base import ElasticLines
 
 
 class GeneticCode(SkbioObject):
-    """Class to hold codon to amino acid mapping, and vice versa.
+    """Genetic code for translating codons to amino acids.
 
     Parameters
     ----------
-    code_sequence : str
-        64-character string containing NCBI representation.
-    id : str, optional
-        identifier for the object.
+    amino_acids : consumable by ``skbio.Protein`` constructor
+        64-character vector containing IUPAC amino acid characters, ordered
+        UUU -> GGG (NCBI's ordering). This is the "AAs" field in NCBI's genetic
+        code format.
+    starts : consumable by ``skbio.Protein`` constructor
+        64-character vector containing only M and - characters, with start
+        codons indicated with M. This is the "Starts" field in NCBI's genetic
+        code format.
     name : str, optional
-        name for the object.
-    start_codon_sequence : str, optional
-        starting point for the codon sequence.
+        Genetic code name. This is simply metadata and does not affect the
+        functionality of the genetic code itself.
 
-    Returns
-    -------
-    GeneticCode
-        initialized ``GeneticCode`` object.
+    See Also
+    --------
+    RNA.translate
+    DNA.translate
+    GeneticCode.from_ncbi
 
-    Raises
-    ------
-    GeneticCodeInitError
-        If the length of `code_sequence` is different to `64`.
+    Notes
+    -----
+    The genetic codes available via ``GeneticCode.from_ncbi`` and used
+    throughout the examples are defined in [1]_. The genetic code strings
+    defined there are directly compatible with the ``GeneticCode`` constructor.
+
+    References
+    ----------
+    .. [1] http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi
```

Comment:
Ok, this one (http://www.ncbi.nlm.nih.gov/BLAST/) isn't either, so maybe the problem is on their side. Just double check that it's working again some time later.